docs/source/final_attrs.rst

```diff
@@ -0,0 +1,267 @@
+Final names, methods and classes
+================================
+
+You can declare a variable or attribute as final, which means that the variable
+must not be assigned a new value after initialization. This is often useful for
+module and class level constants as a way to prevent unintended modification.
+Mypy will prevent further assignments to final names in type-checked code:
+
+.. code-block:: python
+
+   from typing_extensions import Final
+
+   RATE: Final = 3000
+   class Base:
+       DEFAULT_ID: Final = 0
+
+   # 1000 lines later
+
+   RATE = 300  # Error: can't assign to final attribute
+   Base.DEFAULT_ID = 1  # Error: can't override a final attribute
+
+Another use case for final attributes is where a user wants to protect certain
+instance attributes from overriding in a subclass:
+
+.. code-block:: python
+
+   import uuid
+   from typing_extensions import Final
+
+   class Snowflake:
+       """An absolutely unique object in the database"""
+       def __init__(self) -> None:
+           self.id: Final = uuid.uuid4()
+
+   # 1000 lines later
+
+   class User(Snowflake):
+       id = uuid.uuid4()  # Error: can't override a final attribute
+
+Some other use cases might be solved by using ``@property``, but note that
+neither of the above use cases can be solved with it.
+
+.. note::
+
+   This is an experimental feature. Some details might change in later
+   versions of mypy. The final qualifiers are available in the
+   ``typing_extensions`` package available on PyPI.
+
+Syntax variants
+***************
+
+The ``typing_extensions.Final`` qualifier indicates that a given name or
```

Comment:
So I'm not sure that "qualifier" is the best term here, though I'm not sure that "pseudo-type" is any better. We probably should use the same word to describe "ClassVar" (see #5733).